src/objects.c

```diff
@@ -91,7 +91,10 @@ static void construct_client(void *obj)
 	memset(client, 0, sizeof(PgSocket));
 	list_init(&client->head);
 	sbuf_init(&client->sbuf, client_proto);
+	client->packet_cb_state.complete_pkt = malloc(sizeof(struct MBuf));
+	client->packet_cb_state.pkt_header = malloc(sizeof(struct PktHdr));
```

Comment:
I'm playing around with prepared statements larger than pkt_buf to see how badly it breaks. And this was one item I ran into, especially because these allocations are not zero-ed out.

Reply:
Makes sense,
Replace with embedded component